test/e2e/node/kubelet.go

```diff
@@ -479,62 +484,215 @@ var _ = SIGDescribe("kubelet", func() {
 
 		/*
 			Test if kubectl node-logs <node-name>
-			returns something or not!
+			returns an error or not!
 		*/
 
-		ginkgo.It("should return the logs ", func() {
+		ginkgo.It("should return the error of running without --query option", func() {
 			ginkgo.By("Starting the command")
 			tk := e2ekubectl.NewTestKubeconfig(framework.TestContext.CertDir, framework.TestContext.Host, framework.TestContext.KubeConfig, framework.TestContext.KubeContext, framework.TestContext.KubectlPath, ns)
 
 			for nodeName := range nodeNames {
 				cmd := tk.KubectlCmd("alpha", "node-logs", nodeName)
-				runKubectlCommand(cmd, "")
+				_, _, err := framework.StartCmdAndStreamOutput(cmd)
+				if err != nil {
+					framework.Failf("Failed to start kubectl command: %v", err)
+				}
+				err = cmd.Wait()
+				if err == nil {
+					framework.Failf("Command failed: %v. It was expected an error!", cmd)
+				}
 			}
 		})
 
 		/*
-			Test if kubectl node-logs <node-name> --service kubelet
-			returns something or not!
+			Test if kubectl node-logs <node-name> --query kubelet
+			returns the kubelet logs or not!
 		*/
 
-		ginkgo.It("should return the logs for the requested service", func() {
+		ginkgo.It("should return the logs for the requested query", func() {
 			ginkgo.By("Starting the command")
 			tk := e2ekubectl.NewTestKubeconfig(framework.TestContext.CertDir, framework.TestContext.Host, framework.TestContext.KubeConfig, framework.TestContext.KubeContext, framework.TestContext.KubectlPath, ns)
 
 			for nodeName := range nodeNames {
-				cmd := tk.KubectlCmd("alpha", "node-logs", nodeName, "--service", "kubelet")
-				runKubectlCommand(cmd, "--service")
+				cmd := tk.KubectlCmd("alpha", "node-logs", nodeName, "--query", "kubelet")
+				result := runKubectlCommand(cmd, "--query")
+				lookForString("kubelet", result)
 			}
 		})
 
 		/*
-			Test if kubectl node-logs <node-name> --path pods
-			returns something or not!
+			Test if kubectl node-logs --role master --query docker --tail 4
+			returns 4 lines from the docker logs or not!
```

Comment:
hm maybe we should do `containerd` or `crio` depending on which the node is running now that dockershim support is dropped

Reply:
or rather, maybe we should just use `kubelet` as that's guaranteed to be running, and the test isn't testing whether we can get specific logs from the runtime, but that we get the tail of the logs in question

Reply:
Changed it to kubelet. Thank you for pointing this out!